apps/05Mastermind.js

```diff
@@ -23,15 +23,53 @@ function generateSolution() {
 }
 
 function getRandomInt(min, max) {
-  return Math.floor(Math.random() * (max - min)) + min;
+    return Math.floor(Math.random() * (max - min)) + min;
 }
 
-function generateHint() {
+function generateHint(solution, guess) {
     // your code here
+    var solutionArray = solution.split('');
+    var guessArray = guess.split('');
+
+    var correctLetterLocations = 0;
+    var correctLetters = 0;
+
+    for (var i = 0; i < 4; i++) {
+        if (solutionArray[i] === guessArray[i]) {
+            correctLetterLocations++;
+            solutionArray[i] = null;
+        }
+    }
+
+    for (var i = 0; i < 4; i++) {
```

Comment:
There is no need for two for loops here. You are never using the iterator 'i' from the outer for loop, so that's a good indicator that is is unnecessary. You can thus remove the outer for loop and just have:

`
 for (var j = 0; j < 4; j++) {
         if (solutionArray.indexOf(guessArray[j]) !== -1) {
                 correctLetters++;
             }
         }
`